Gems/Atom/Tools/ShaderManagementConsole/Code/Source/ShaderManagementConsoleApplication.cpp

```diff
@@ -248,4 +251,71 @@ namespace ShaderManagementConsole
 
         return shaderItems;
     }
+
+    AZStd::vector<AZ::Data::AssetId> ShaderManagementConsoleApplication::GetAllMaterialAssetIds()
+    {
+        AZStd::vector<AZ::Data::AssetId> assetIds;
+
+        AZ::Data::AssetCatalogRequests::AssetEnumerationCB collectAssetsCb =
+            [&]([[maybe_unused]] const AZ::Data::AssetId id, const AZ::Data::AssetInfo& info)
+        {
+            if (info.m_assetType == AZ::RPI::MaterialAsset::RTTI_Type())
+            {
+                assetIds.push_back(id);
+            }
+        };
+
+        AZ::Data::AssetCatalogRequestBus::Broadcast(
+            &AZ::Data::AssetCatalogRequestBus::Events::EnumerateAssets, nullptr, collectAssetsCb, nullptr);
+
+        return assetIds;
+    }
+
+    AZStd::string ShaderManagementConsoleApplication::GetFullSourcePathFromRelativeProductPath(const AZStd::string& relativeProductPath)
+    {
+        AZStd::string fullSourcePath;
+        bool fullPathFound = false;
+        AzToolsFramework::AssetSystemRequestBus::BroadcastResult(
+            fullPathFound,
+            &AzToolsFramework::AssetSystem::AssetSystemRequest::GetFullSourcePathFromRelativeProductPath,
+            relativeProductPath,
+            fullSourcePath);
+
+        if (fullPathFound)
+        {
+            return fullSourcePath;
+        }
+        else
+        {
+            AZ_Error(
+                "GetFullSourcePathFromRelativeProductPath",
+                false,
+                "Failed to get full sorece path for relative product path %s.",
+                relativeProductPath.c_str());
+        }
+
+        return "";
+    }
+
+    AZStd::string ShaderManagementConsoleApplication::GenerateRelativeSourcePath(const AZStd::string& fullShaderPath)
+    {
+        bool pathFound = false;
+        AZStd::string relativePath, rootFolder;
+        AzToolsFramework::AssetSystemRequestBus::BroadcastResult(
+            pathFound,
+            &AzToolsFramework::AssetSystemRequestBus::Events::GenerateRelativeSourcePath,
+            fullShaderPath,
+            relativePath,
+            rootFolder);
```

Comment:
Ideally this ought to be exposed from the asset system's python API rather than ShaderManagementConsole.
(@o3de/sig-core )